frontend/packages/topology/src/components/quick-search/utils/quick-search-utils.ts

```diff
@@ -0,0 +1,29 @@
+import { CatalogItem } from '@console/plugin-sdk';
+import { keywordCompare } from '@console/dev-console/src/components/catalog/utils/catalog-utils';
+import { QuickStart } from '@console/app/src/components/quick-starts/utils/quick-start-types';
+
+export const quickSearch = (items: CatalogItem[], query: string) => {
+  return items.filter((item) => keywordCompare(query, item));
+};
+
+export const getTransformedQuickStarts = (
+  quickStarts: QuickStart[],
+  setActiveQuickStart: (quickStartId: string, totalTasks?: number) => void,
+): CatalogItem[] => {
+  return quickStarts.map((qs: QuickStart) => ({
+    name: qs.spec.displayName,
+    type: 'Quick Start',
+    uid: qs.metadata.uid,
+    cta: {
+      callbackfn: () => setActiveQuickStart(qs.metadata.name, qs.spec.tasks?.length),
```

Comment:
This is because we do not instantiate states for all the quick starts on load. We do that lazily when a quick start is actually opened, that's why we need to pass the number of tasks here. We can look at improving this for all quick starts in a separate issue and instantiate quick starts states eagerly on quick start load.

Reply:
This is very odd that we need this information to init the task state to `INIT`. Can we simply assume that a task without state is in the initial state?

Reply:
I don't exactly remember what caused me to implement it this way but maybe we can update this to assume the initial state.